src/assembler.cpp

```diff
@@ -1205,13 +1209,32 @@ int main(int argc, char* argv[])
                 constant.value_hex[1] = 0;
                 constant.value_hex[2] = 0;
                 constant.value_hex[3] = 0;
-                constant.type = ConstantInfo::Float;
                 constant.regid = LookupIdentifier(id).GetIndex();
 
-                constant.f.x = to_float24(values[0]);
-                constant.f.y = to_float24(values[1]);
-                constant.f.z = to_float24(values[2]);
-                constant.f.w = to_float24(values[3]);
+
+
+                switch (ret.GetType()) {
+                    case RegisterType::FloatUniform: {
```

Comment:
Please use this style instead:

    case RegisterType::FloatUniform:
    {
        constant.....
        break;
    }